iot_relay_server/app/ai_agent.py

```
@@ -54,237 +63,273 @@
+            if new != 0:
+                return True
             continue
-        if abs(new - old) / abs(old) > 0.05: return True
+        if abs(new - old) / abs(old) > 0.05:
+            return True
     old_soil = _last_sensor_data.get("soil_moisture") or 50
     new_soil = new_data.get("soil_moisture") or 50
-    if abs(new_soil - old_soil) > 3: return True
+    if abs(new_soil - old_soil) > 3:
+        return True
     return False
```

Ruff (0.15.9)

[warning] Return the condition `abs(new_soil - old_soil) > 3` directly

Replace with `return abs(new_soil - old_soil) > 3`

(SIM103)